src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,83 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+/**
+ * Represents Task that the user wants Duke to remember
+ */
+public abstract class Task {
+
+    protected static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy");
+    protected static final DateTimeFormatter SAVE_DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    protected static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+
+    private final String name;
```

Comment:
The book says: "A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail."
perhaps you can change the "name" variable to a more meaning name, such as taskName.